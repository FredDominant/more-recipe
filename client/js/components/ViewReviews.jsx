import React from 'react';
import PropTypes from 'prop-types';

const ViewReviews = props => (
  <div className="container">
    <div className="row">
      <div className="col-sm-3">
        <div className="reviewer-image">
          <img src="" alt="" height="100" width="20" className="img-rounded" />
        </div>
      </div>
      <div className="col-sm-9">
        <div className="content">
          <h6> {props.content} </h6>
          <h6> - <small>{`${props.firstname} ${props.lastname}`} </small> </h6>
        </div>
      </div>
    </div>
    <hr />
  </div>
);

ViewReviews.propTypes = {
  content: PropTypes.string.isRequired,
  firstname: PropTypes.string.isRequired,
  lastname: PropTypes.string.isRequired
};

export default ViewReviews;
