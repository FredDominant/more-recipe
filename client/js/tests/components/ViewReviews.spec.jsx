import React from 'react';
import { shallow } from 'enzyme';

import ViewReviews from '../../components/ViewReviews';
/**
 * @description test for Body component
 *
 * @returns {null} null
 */
function setup() {
  const wrapper = shallow(<ViewReviews />);

  return {
    wrapper
  };
}

describe('Test for ViewReviews component', () => {
  it('should render correctly', () => {
    const { wrapper } = setup();
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('div').length).toBeGreaterThan(0);
    expect(wrapper.find('img').length).toBe(1);
    expect(wrapper.find('.reviewer-name').length).toBe(1);
    expect(wrapper.find('#reviewer-image').length).toBe(1);
    expect(wrapper.find('.reviewer-image').length).toBe(1);
    expect(wrapper.find('small').length).toBe(3);
    expect(wrapper.find('#review-date').length).toBe(1);
    expect(wrapper.find('.row').length).toBe(1);
  });
});
