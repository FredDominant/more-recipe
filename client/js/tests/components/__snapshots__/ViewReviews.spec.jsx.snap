// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for ViewReviews component should render correctly 1`] = `
<div
  className=""
>
  <div
    className="row"
  >
    <div
      className="col-xs-1 col-sm-1 col-md-1 col-lg-1"
    >
      <div
        className="reviewer-image"
      >
        <img
          alt=""
          className="rounded float-left img-fluid"
          id="reviewer-image"
          src=""
        />
      </div>
    </div>
    <div
      className="col-xs-10 col-sm-10 col-md-10 col-lg-10"
    >
      <div
        className="content text-left mb-5"
      >
        <h5
          className="reviewer-name"
        >
           
          <small>
             
             
             
          </small>
           
        </h5>
        <h6>
           
          <small>
             
          </small>
        </h6>
        <h6>
           
          <small
            id="review-date"
          >
             
          </small>
           
        </h6>
      </div>
    </div>
  </div>
</div>
`;
